test/integration/routing.go

```diff
@@ -174,3 +193,48 @@ func (t *routing) verifyFaultInjection(src, dst, headerKey, headerVal string,
 	}
 	return nil
 }
+
+// verifyRedirect verifies if the http redirect was setup properly
+func (t *routing) verifyRedirect(src, dst, targetHost, targetPath, headerKey, headerVal string, respCode int) error {
+	url := fmt.Sprintf("http://%s/%s", dst, src)
+	glog.Infof("Making 1 request (%s) from %s...\n", url, src)
+
+	resp := t.clientRequest(src, url, 1, fmt.Sprintf("-key %s -val %s", headerKey, headerVal))
+
+	statusCode := ""
+	if len(resp.code) > 0 {
+		statusCode = resp.code[0]
+	}
+
+	if strconv.Itoa(respCode) != statusCode {
+		return fmt.Errorf("redirect verification failed: "+
+			"response status code: %v, expected %v",
+			statusCode, respCode)
+	}
+
+	exp := regexp.MustCompile("(?i)Host=(.*)")
+	hosts := exp.FindAllStringSubmatch(resp.body, -1)
+	host := ""
+	if len(hosts) > 0 {
+		host = hosts[0][1]
+	}
+	if host != targetHost {
+		return fmt.Errorf("redirect verification failed: "+
+			"response body contains Hostname=%v, expected Hostname=%v",
+			hosts[0], targetHost)
+	}
+
+	exp = regexp.MustCompile("(?i)URL=(.*)")
+	paths := exp.FindAllStringSubmatch(resp.body, -1)
```

Comment:
in this case however there are two occurrences of "URL" in the response, the original URL and the redirected URL